### PR TITLE
Composer install script

### PR DESCRIPTION
**Pull request by manuerumx**
For: https://github.com/Medology/General-Issues/issues/106

Composer installation script refactor.

[Original PR](https://github.com/Medology/iLab/pull/808) was created to validate the refactor and ensure CI execute the changes without errors

The new method uses the provided sha256sum files by [composer](https://getcomposer.org/download/) to validate the existing and downloaded files match with the expected version.

### CODE REVIEW

File: bin/init/install-composer.sh [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+#############################################
+# Validate the sha256sum against the existing composer file
+#
+# Globals:
+#   $ROOT root project directory
+#
+# Returns:
+#   "composer: OK" when the validation pass otherwise will return different type of errors
```
Comment (jose-ochoa) [inline bin/init/install-composer.sh:39]: please update the return description to reflect the 1/0 status code

File: bin/init/install-composer.sh [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+# Download the sha256-sum file from composer page for the expected version
+#
+# Globals:
+#   $EXPECTED_VERSION the expected composer version to install/validate
+#   $ROOT             root project directory
+#
+#############################################
+downloadComposerSHA() {
+  # Download the original sha256 sum for the expected version
+  curl -L https://getcomposer.org/download/${EXPECTED_VERSION}/composer.phar.sha256sum > "$ROOT"/bin/composer.sha256sum
```
Comment (jose-ochoa) [inline bin/init/install-composer.sh:19]: I don't think we should download the SHA, we should hard code it like the expected version. And check it against the download file. If they were to overwrite the file and update the SHA we wouldn't know

**Review by jose-ochoa — CHANGES_REQUESTED** [review]

File: bin/init/install-composer.sh (hunk 2 above)
Comment (manuerumx) [reply bin/init/install-composer.sh:19]: I don't see why the SHA could change unless it's a different file and most likely, a new version.

What you suggest is the way we check the composer installation. So far I'm not aware of the behavior you are worried about ever happening before. 

If the composer organization changes the file and the SHA, but the changes are published with the same version, the PHP community has a bigger problem.

I think the idea of these validations is to ensure the downloaded file is not corrupted by a problem during its download.

Next week, we can discuss this with the team.

Comment (jose-ochoa) [reply bin/init/install-composer.sh:19]: Verifying against a known SHA and explicitly comparing them is the only guarantee we can have, it's like the composer.lock file equivalent. But I'll comply with whatever outcome is.

Comment (rashidsaf) [reply bin/init/install-composer.sh:19]: The chance of this happening is highly unlikely.

**Review by rashidsaf — APPROVED** [review]

**Comment by manuerumx** [comment]
Are you ok with the changes @jose-ochoa

File: bin/init/install-composer.sh [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+#############################################
+# Validate the sha256sum against the existing composer file
+#
+# Globals:
+#   $ROOT root project directory
+#
+# Returns:
+#   1 when the validation pass 
+#   0 when the sha validation fail
+#s
```
Comment (jose-ochoa) [inline bin/init/install-composer.sh:41]: Please remove cmd + `s`

**Review by jose-ochoa — CHANGES_REQUESTED** [review]

File: bin/init/install-composer.sh (hunk 2 above)
Comment (jose-ochoa) [reply bin/init/install-composer.sh:19]: image

**Review by jose-ochoa — APPROVED** [review]

**Pull request merged**